Sources/NIOCore/Docs.docc/loops-futures-concurrency.md

```diff
@@ -0,0 +1,176 @@
+# EventLoops, EventLoopFutures, and Swift Concurrency
+
+This article aims to communicate how NIO's ``EventLoop``s and ``EventLoopFuture``s interact with the Swift 6
+concurrency model, particularly regarding data-race safety. It aims to be a reference for writing correct
+concurrent code in the NIO model.
+
+NIO predates the Swift concurrency model. As a result, several of NIO's concepts are not perfect matches to
+the concepts that Swift uses, or have overlapping responsibilities.
+
+## Isolation domains and executors
+
+First, a quick recap. The core of Swift 6's data-race safety protection is the concept of an "isolation
+domain". Some valuable reading regarding the concept can be found in
+[SE-0414 (Region based isolation)](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0414-region-based-isolation.md)
+but at a high level an isolation domain can be understood to be a region within which there cannot be
```

Comment:
I wouldn't use the word "region" here because that's easily confusable with "isolation region" which is a completely different thing

Reply:
Reworded.